Applications/Utils/FileConverter/GMSH2OGS.cpp

```diff
@@ -111,6 +116,99 @@ int main (int argc, char* argv[])
             INFO("Mesh does not contain any lines.");
         }
     }
+    // *** print meshinformation
+    INFO("Please check your mesh carefully!");
+    INFO(
+        "Degenerated or redundant mesh elements can cause OGS to stop or "
+        "misbehave.");
+    INFO("Use the -e option to delete redundant line elements.");
+    // Geometric information
+    const GeoLib::AABB aabb(MeshLib::MeshInformation::getBoundingBox(*mesh));
+    auto minPt(aabb.getMinPoint());
+    auto maxPt(aabb.getMaxPoint());
+    INFO("Node coordinates:");
+    INFO("\tx [%g, %g] (extent %g)", minPt[0], maxPt[0], maxPt[0] - minPt[0]);
+    INFO("\ty [%g, %g] (extent %g)", minPt[1], maxPt[1], maxPt[1] - minPt[1]);
+    INFO("\tz [%g, %g] (extent %g)", minPt[2], maxPt[2], maxPt[2] - minPt[2]);
+
+    INFO("Edge length: [%g, %g]", mesh->getMinEdgeLength(),
+         mesh->getMaxEdgeLength());
+
+    // Element information
+    const std::array<unsigned, 7> nr_ele_types(
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh));
+    INFO("Element types:");
+    unsigned etype = 0;
+    if (nr_ele_types[etype] > 0)
+    {
+        INFO("\t%d lines", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d triangles", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d quads", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d tetrahedra", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d hexahedra", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d pyramids", nr_ele_types[etype]);
+    }
+    if (nr_ele_types[++etype] > 0)
+    {
+        INFO("\t%d prisms", nr_ele_types[etype]);
+    }
```

Comment:
something like a table would be easier to understand.
```c++
std::array element_names {"lines", ...};
for (int i = 0; i < nr_ele_types.size(); ++i)
{
    INFO("\t%d %s", nr_ele_types[i], element_names[i]);
}
```

Reply:
@TomFischer This parts were copied from checkMesh.cpp. Where could we move this output routines?

Reply:
The array returned by `MeshLib::MeshInformation::getNumberOfElementTypes` is only used for the output. Since this output is directly related to the `MeshInformation` I would put it in the class `MeshInformation`. Furthermore, the lines of code used for the output can reduced by a for-loop, as already suggested:
```c++
for (int element_type = 0; element_type < nr_ele_types.size();
         element_type++)
    {
        INFO("\t%d %s", nr_ele_types[element_type],
             MeshLib::MeshElemType2String(
                 static_cast<MeshLib::MeshElemType>(element_type)).c_str());
    }
```